torch/csrc/jit/script/python_type_parser.cpp

```diff
@@ -67,19 +67,18 @@ subscript_to_type_fns() {
                  parseTypeFromExpr(*subscript.subscript_exprs().begin());
              return FutureType::create(elem_type);
            }},
-           {"Dict",
-            [](Subscript subscript) -> TypePtr {
-              if (subscript.subscript_exprs().size() != 2) {
-                throw ErrorReport(subscript)
-                    << " expected exactly 2 element types but found "
-                    << subscript.subscript_exprs().size();
-              }
-              auto key_type =
-                  parseTypeFromExpr(subscript.subscript_exprs()[0]);
-              auto value_type =
-                  parseTypeFromExpr(subscript.subscript_exprs()[1]);
-              return DictType::create(key_type, value_type);
-            }},
+          {"Dict",
```

Comment:
I don't thing the name 'python_type_parser' is clear because both of the files are effective parsing python-like syntax. This will become more confusing in the future when Schema types are moved over to python syntax. To be more future proof in the naming maybe call the one used in schema `schema_type_parser` and the other one used in the TorchScript compiler `script_type_parser`